server/opts.go

```diff
@@ -615,6 +660,206 @@ func parseCluster(v interface{}, opts *Options, errors *[]error, warnings *[]err
 	return nil
 }
 
+func parseURLs(a []interface{}, typ string) ([]*url.URL, []error) {
+	var (
+		errors []error
+		urls   = make([]*url.URL, 0, len(a))
+	)
+	for _, u := range a {
+		tk, u := unwrapValue(u)
+		sURL := u.(string)
+		url, err := parseURL(sURL, typ)
+		if err != nil {
+			err := &configErr{tk, err.Error()}
+			errors = append(errors, err)
+			continue
+		}
+		urls = append(urls, url)
+	}
+	return urls, errors
+}
+
+func parseURL(u string, typ string) (*url.URL, error) {
+	urlStr := strings.TrimSpace(u)
+	url, err := url.Parse(urlStr)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing %s url [%q]", typ, urlStr)
+	}
+	return url, nil
+}
+
+func parseGateway(v interface{}, o *Options, errors *[]error, warnings *[]error) error {
+	tk, v := unwrapValue(v)
+	gm, ok := v.(map[string]interface{})
+	if !ok {
+		return &configErr{tk, fmt.Sprintf("Expected gateway to be a map, got %T", v)}
+	}
+	for mk, mv := range gm {
+		// Again, unwrap token value if line check is required.
+		tk, mv = unwrapValue(mv)
+		switch strings.ToLower(mk) {
+		case "name":
+			o.Gateway.Name = mv.(string)
+		case "listen":
+			hp, err := parseListen(mv)
+			if err != nil {
+				err := &configErr{tk, err.Error()}
+				*errors = append(*errors, err)
+				continue
+			}
+			o.Gateway.Host = hp.host
+			o.Gateway.Port = hp.port
+		case "port":
+			o.Gateway.Port = int(mv.(int64))
+		case "host", "net":
+			o.Gateway.Host = mv.(string)
+		case "authorization":
```

Comment:
Not blocking but let's discuss again this week on future plans for this and routes.